samples/scenarios/jdbc_kerberos/README.md

```diff
@@ -0,0 +1,152 @@
+
+In simple words - SSPI is used to configure Single Sign-on type of authentication to connect to a database using your Active Directory credentials. Tableau Desktop uses [native GSS-API](https://docs.oracle.com/en/java/javase/11/security/accessing-native-gss-api.html) interface for SSPI/SSO using Kerberos.
+
+#### JAAS and Java GSS-API
+
+JAAS(Java Authentication and Authorization Service) can be used for two purposes:
+
+-   for authentication of users, to reliably and securely determine who is currently executing Java code, and
+-   for authorization of users to ensure they have the access control rights (permissions) required to do security-sensitive operations.
+
+Java GSS-API is used for securely exchanging messages between communicating applications. The Java GSS-API contains the Java bindings for the GSS-API defined in [RFC 5653](https://tools.ietf.org/html/rfc5653). 
+(source - [Oracle Java docs](https://docs.oracle.com/en/java/javase/11/security/java-generic-security-services-java-gss-api1.html)).  
+
+JAAS authentication is typically performed prior to secure communication using Java GSS-API. Tableau Server uses JAAS and JGSS for Server Runas and Kerberos Delegation authentication.
+
+## <a id="desktop"></a>Kerberos SSO for Tableau Desktop
+
+### SSPI authentication for Tableau Desktop on Windows
+connection-fields.xml
+```
+<field name="authentication" label="Authentication" category="authentication" value-type="selection">
+  <selection-group>
+    <option value="auth-integrated" label="Integrated" />
+    <!-- other auth options -->
+  </selection-group>
+</field>
+```
+connectionProperties.js
+```
+    props["gsslib"] = "gssapi";	 
+    props["jaasLogin"] = "false";  
+    props["jaasApplicationName"] = "com.sun.security.jgss.krb5.initiate";
+```
+
+![Image](images/DesktopConnectionDialog.png)
+
+### Kerberos SSO for Tableau Desktop on Mac
+connection-fields.xml
+```
+<field name="authentication" label="Authentication" category="authentication" value-type="selection">
+  <selection-group>
+    <option value="auth-integrated" label="Integrated" />
+    <!-- other auth options -->
+  </selection-group>
+</field>
+```
+
+connectionProperties.js
+```
+    props["jaasApplicationName"] = "com.sun.security.jgss.krb5.initiate";
+    props["gsslib"] = "gssapi";	 
+    props["jaasLogin"] = "false";  
+```
+
+Other Configuration:
+* Only for postgres jdbc plugin on Mac - add this property `Settings.DisableNativeGSS` as a `Boolean/YES` to the Tableau plist on Mac at `/Library/Preferences/com.tableau.Tableau-<version>.plist` as described below. Here version is Tableau Desktop major version, for example, `com.tableau.Tableau-2020.4.plist`.
+```
+// Note - the below command requires root privileges hence sudo is needed.
+sudo defaults write /Library/Preferences/com.tableau.Tableau-<version>.plist Settings.DisableNativeGSS -bool YES
```

Comment:
Did you ever open a bug on Postgres JDBC for that?

Reply:
I did not open a bug on them. I should probably do that. This is not a great experience.

Reply:
yes, I almost forgot about that. This is the [issue](https://github.com/pgjdbc/pgjdbc/issues/1662) which I reported some time back with 42.2.5 driver. Now we are using 42.2.14 and it is still not been fixed.